### ISSUE TITLE
Exposing widget tree through serviceManager

### ISSUE THREAD

**Issue by JoseAlba** [flutter/devtools#8150]
Hello!

I have been working on a dart dev tool extension for my application but I am having difficulty getting access to the widget tree of the app that I am running. I tried getting access to it through the `serviceManager` in `DevToolsExtension` but I wasn't able to successfully. 

I see that in [inspector_v2](https://github.com/flutter/devtools/tree/master/packages/devtools_app/lib/src/screens/inspector_v2) it is able to get access to the rootTree through the InspectorController. I was wondering if it would be possible to expose this API so that developers could easily navigate through the tree and make dart dev tool extensions.

Currently the only way I can think of doing this is through exposing the root widget through a logger or platform channel that could be used as a median through the app and dart devtool extension.

If this is already implemented I was hoping you could share the code snippet so that I can add it to my package.

**Comment by kenzieschmoll** [comment]
What did you try for this? This should be possible by calling the same service extension that DevTools Flutter Inspector [calls](https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart/#L968), [getRootWidgetTree](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/service_extensions.dart/#L394).
```dart
serviceManager.service.callServiceExtension(
  'ext.flutter.inspector.${WidgetInspectorServiceExtensions.getRootWidgetTree.name}'
  serviceManager.isolateManager.mainIsolate.value.id,
  args: {
    'groupName': 'myExtensionGroup', // *see question about this below
    'isSummaryTree': <'true' or 'false'> // true if you only want widgets created in the user's code
    'withPreviews': <'true' or 'false'> // true 
  },
);
```

* @elliette should `groupName` actually be optional? The groupName is passed along to `InspectorSerializationDelegate` [here](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/widget_inspector.dart/#L2027), but in `InspectorSerializationDelegate` the groupName is [optional](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/widget_inspector.dart/#L3762).

**Comment by elliette** [comment]
> @elliette should groupName actually be optional? The groupName is passed along to InspectorSerializationDelegate [here](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/widget_inspector.dart/#L2027), but in InspectorSerializationDelegate the groupName is [optional](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/widget_inspector.dart/#L3762).

Good point! I think it can be optional. Opened https://github.com/flutter/devtools/issues/8199 to track.

**Comment by kenzieschmoll** [comment]
Closing this since calling the service extension directly will provide the widget tree.
